server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorCleanupOvershadowed.java

```diff
@@ -82,6 +83,19 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
             stats.addToGlobalStat("overShadowedCount", 1);
           }
         }
+
+        for (LoadQueuePeon loadQueue : coordinator.getLoadManagementPeons().values()) {
+          for (DataSegment dataSegment: loadQueue.getSegmentsToLoad()) {
+            timeline = timelines.get(dataSegment.getDataSource());
+            if (timeline == null) continue;
+            timeline.add(dataSegment.getInterval(), dataSegment.getVersion(), dataSegment.getShardSpec().createChunk(dataSegment));
```

Comment:
can you add a comment here explaining why this is in its own loop? something like 

```java
// Temporarily add queued segments to the timeline to see if they still need to be loaded.
```